lib/clearance/user.rb

```diff
@@ -75,13 +75,13 @@ def reset_remember_token!
     def update_password(new_password)
       self.password_changing = true
       self.password = new_password
+      valid?
 
-      if valid?
+      if errors[:password].blank?
         self.confirmation_token = nil
         generate_remember_token
+        save(validate: false)
```

Comment:
I'm in support of anything getting engine-specific stuff out of my User.

If we don't go that route, I just wanted to note that `update_columns` won't touch `update_at`, which is a good or bad thing depending on how you look at it.